pkg/cvo/sync_worker.go

```diff
@@ -636,6 +638,26 @@ func (w *SyncWorker) Start(ctx context.Context, maxWorkers int, cvoOptrName stri
 				utilruntime.HandleError(fmt.Errorf("unable to synchronize image (waiting %s): %v", interval, err))
 				continue
 			}
+			// Set failing condition for report only errors.
+			if err != nil {
+				config, cvErr := lister.Get(cvoOptrName)
+				if cvErr != nil {
+					klog.V(2).Infof("Could not set ClusterStatusFailing condition. Error getting ClusterVersion, err=%v", cvErr)
+				} else {
+					var reason string
+					if uErr, ok := err.(*payload.UpdateError); ok {
+						reason = uErr.Reason
+					}
+					// set the failing condition
+					resourcemerge.SetOperatorStatusCondition(&config.Status.Conditions, configv1.ClusterOperatorStatusCondition{
+						Type:               ClusterStatusFailing,
```

Comment:
This makes me a bit jumpy, because while `pkg/cvo/availableupdates.go` touches `RetrievedUpdates` (under `availableUpdates.NeedsUpdate`, called from `pkg/cvo/status.go`) and `pkg/cvo/upgradeable.go` touches `Upgradeable`) (under `upgradeable.NeedsUpdate`, also called from `pkg/cvo/status.go`), this one is reaching out to `lister`.  And then it's... not clear to me how the resulting value is making its way back into etcd.  Or maybe, it's not clear to me why the `convertErrorToProgressing` handling in `pkg/cvo/status.go` [isn't getting `Failing` set here][1].

But your `bool` return or an `UpdateEffect` cast and `UpdateEffectReport` check, or similar should be enough to cover [the "failing to complete the install" aspect][1].

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1951835#c23

Reply:
Yeah. Got rid of that hack and replaced with `cr.CompleteWithErrors(reportEffectErrors)` in `apply`